compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -891,9 +899,10 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                                 trait_ref: ty::TraitRef::new(
                                     self.tcx,
                                     inner.trait_ref.def_id,
-                                    self.tcx.mk_args(
-                                        &[&[l_ty.into(), r_ty.into()], &inner.trait_ref.args[2..]]
-                                            .concat(),
+                                    self.tcx.mk_args_trait(
```

Comment:
Why did you use `mk_args_trait` here? It's harder to read, imo. You should just use `mk_args` like before. The indexing that was previous here should be correct if we're always using binop traits here.

Reply:
Mainly due to the indexing problem--I reverted it to `mk_args`.